### PR TITLE
Prepare interop driver for a metadata-only mode.

### PR DESCRIPTION
**Pull request by sbogolepov**

### CODE REVIEW

File: utilities/src/main/kotlin/org/jetbrains/kotlin/cli/utilities/InteropCompiler.kt [hunk 1]
```diff
@@ -69,6 +69,10 @@ fun invokeInterop(flavor: String, args: Array<String>): Array<String> {
     }
 
     val cinteropArgsToCompiler = interop(flavor, args + additionalArgs, additionalProperties)
+    // There is no need in compiler invocation if we're generating only metadata.
```
Comment (SvyatoslavScherbina) [inline utilities/src/main/kotlin/org/jetbrains/kotlin/cli/utilities/InteropCompiler.kt]: Why not return `null` from `interop(...)` above instead?

File: Interop/StubGenerator/src/main/kotlin/org/jetbrains/kotlin/native/interop/gen/jvm/main.kt [hunk 2]
```diff
@@ -270,31 +278,48 @@ private fun processCLib(args: Array<String>, additionalArgs: Map<String, Any> =
...
+            val compilerCmd = arrayOf(compiler, *compilerArgs,
+                    "-c", outCFile.absolutePath, "-o", outOFile.absolutePath)
+            runCmd(compilerCmd, verbose)
+
+            val outLib = File(nativeLibsDir, System.mapLibraryName(libName))
+            val linkerCmd = arrayOf(linker,
+                    outOFile.absolutePath, "-shared", "-o", outLib.absolutePath,
+                    *linkerOpts)
+            runCmd(linkerCmd, verbose)
+            outOFile.absolutePath
```
Comment (SvyatoslavScherbina) [inline Interop/StubGenerator/src/main/kotlin/org/jetbrains/kotlin/native/interop/gen/jvm/main.kt]: Shouldn't it be more like `outLib.absolutePath`?

Comment (sbogolepov) [reply Interop/StubGenerator/src/main/kotlin/org/jetbrains/kotlin/native/interop/gen/jvm/main.kt]: Yeah, it’s a typo.

File: Interop/StubGenerator/src/main/kotlin/org/jetbrains/kotlin/native/interop/gen/jvm/GenerationMode.kt [hunk 3]
```diff
@@ -0,0 +1,13 @@
+package org.jetbrains.kotlin.native.interop.gen.jvm
```
Comment (olonho) [inline Interop/StubGenerator/src/main/kotlin/org/jetbrains/kotlin/native/interop/gen/jvm/GenerationMode.kt]: Copyright.

**Pull request merged**